core/cache/watcher.go

```diff
@@ -258,3 +259,37 @@ func (w *ChangeWatcher) changed(topic string, value interface{}) {
 
 	w.notify(strings)
 }
+
+// RegexpChangeWatcher notifies when individual pieces of a subscribed topic
+// match the given compiled regular expression.  An initial event is sent
+// with the input given at creation.
+type RegexpChangeWatcher struct {
+	*stringsWatcherBase
+
+	compiled *regexp.Regexp
+}
+
+func newRegexpAddRemoveWatcher(compiled *regexp.Regexp, values ...string) *RegexpChangeWatcher {
+	return &RegexpChangeWatcher{
+		stringsWatcherBase: newStringsWatcherBase(values...),
+		compiled:           compiled,
+	}
+}
+
+func (w *RegexpChangeWatcher) changed(topic string, value interface{}) {
+	strings, ok := value.([]string)
+	if !ok {
+		logger.Errorf("programming error, value not of type []string")
+	}
+
+	matches := set.NewStrings()
+	for _, s := range strings {
+		if w.compiled.MatchString(s) {
```

Comment:
Yeah, this could be a `predicate` to the changes watcher, and you provide high order functions to pass in the changes watcher.

```go
func regexpPredicate(compiled *regexp.Regexp) func(string) bool {
    return func(value string) bool { return compiled.MatchString(value) }
}
func stringPredicate(expected string) func(string) bool {
    return func(value string) bool { return expected == value) }
} 

func newRegexpAddRemoveWatcher(fn predicate, values ...string) *RegexpChangeWatcher {
```

Reply:
This abstraction tends toward the generic generated watcher PR that got no play :(

Reply:
:-(